### DISCUSSION TITLE
Installation in Windows

### DISCUSSION THREAD

**Discussion by fatdennis** [vicwomg/pikaraoke#221] [Q&A]
Hi, I want to install it in Windows for a try.  But when I try to install ms visual C, it found a dead link, and then when I tried "pip3 install -r requirements.txt", so many error come out and then know how to start...Can anyone help?

**Comment by maines-pet** [comment]
Just installed this in my Windows laptop recently. I was also having a problem with installation from the requirements.txt when I have Python 3.10 installed. When I downgraded to Python 3.7, the installation went through.

**Comment by vicwomg** [comment]
I'd be curious to see the error output with python 3.10

**Comment by lsibarra** [comment]
Hi,
I've been trying to install the new pikaraoke version 1.2 and having issues. Can you help? I am running windows 10 and when I launch pikaraoke.bat The chrome browser opens but showing "This site can't be reached" See screenshot. 
![image](https://github.com/vicwomg/pikaraoke/assets/135774001/3e1bedfc-1399-4de6-9e64-e562b008dd4a)

Also below is the error log

`C:\Program Files\pikaraoke-1.2\app.py:617: SyntaxWarning: invalid escape sequence '\S'
  return os.path.join(os.path.dirname(__file__), ".venv\Scripts\yt-dlp.exe")
C:\Program Files\pikaraoke-1.2\app.py:642: SyntaxWarning: invalid escape sequence '\p'
  legacy_directory = os.path.expanduser("~\pikaraoke\songs")
C:\Program Files\pikaraoke-1.2\app.py:646: SyntaxWarning: invalid escape sequence '\p'
  return "~\pikaraoke-songs"
[2024-01-04 18:21:49] INFO: Fetching available songs in: C:\Users\lsiba\pikaraoke-songs/
[04/Jan/2024:18:21:50] ENGINE Bus STARTING
[2024-01-04 18:21:50] INFO: [04/Jan/2024:18:21:50] ENGINE Bus STARTING
[04/Jan/2024:18:21:50] ENGINE Serving on http://0.0.0.0:5555
[2024-01-04 18:21:50] INFO: [04/Jan/2024:18:21:50] ENGINE Serving on http://0.0.0.0:5555
[04/Jan/2024:18:21:50] ENGINE Bus STARTED
[2024-01-04 18:21:50] INFO: [04/Jan/2024:18:21:50] ENGINE Bus STARTED

DevTools listening on ws://127.0.0.1:52780/devtools/browser/9d0953a7-55e4-4b40-9bdd-024054879823
Traceback (most recent call last):
  File "C:\Program Files\pikaraoke-1.2\app.py", line 864, in <module>
    driver.get(f"{k.url}/splash" )
  File "C:\Program Files\pikaraoke-1.2\.venv\Lib\site-packages\selenium\webdriver\remote\webdriver.py", line 357, in get
    self.execute(Command.GET, {"url": url})
  File "C:\Program Files\pikaraoke-1.2\.venv\Lib\site-packages\selenium\webdriver\remote\webdriver.py", line 348, in execute
    self.error_handler.check_response(response)
  File "C:\Program Files\pikaraoke-1.2\.venv\Lib\site-packages\selenium\webdriver\remote\errorhandler.py", line 229, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.WebDriverException: Message: unknown error: net::ERR_NAME_NOT_RESOLVED
  (Session info: chrome=120.0.6099.199)
Stacktrace:
        GetHandleVerifier [0x00007FF698A22142+3514994]
        (No symbol) [0x00007FF698640CE2]
        (No symbol) [0x00007FF6984E76AA]
        (No symbol) [0x00007FF6984DFB39]
        (No symbol) [0x00007FF6984D25DD]
        (No symbol) [0x00007FF6984D16FD]
        (No symbol) [0x00007FF6984D0746]
        (No symbol) [0x00007FF6984D06EB]
        (No symbol) [0x00007FF6984CEE3D]
        (No symbol) [0x00007FF6984CF603]
        (No symbol) [0x00007FF6984EA0FD]
        (No symbol) [0x00007FF698572E01]
        (No symbol) [0x00007FF698555FEA]
        (No symbol) [0x00007FF6985728F6]
        (No symbol) [0x00007FF698555D93]
        (No symbol) [0x00007FF698524BDC]
        (No symbol) [0x00007FF698525C64]
        GetHandleVerifier [0x00007FF698A4E16B+3695259]
        GetHandleVerifier [0x00007FF698AA6737+4057191]
        GetHandleVerifier [0x00007FF698A9E4E3+4023827]
        GetHandleVerifier [0x00007FF6987704F9+689705]
        (No symbol) [0x00007FF69864C048]
        (No symbol) [0x00007FF698648044]
        (No symbol) [0x00007FF6986481C9]
        (No symbol) [0x00007FF6986388C4]
        BaseThreadInitThunk [0x00007FFB92497344+20]
        RtlUserThreadStart [0x00007FFB933226B1+33]
`

**Comment by vicwomg** [comment]
This could be an issue with mDNS on your network.

Try launching with “—prefer-ip” and see if it works better.

**Comment by lsibarra** [comment]
Hi @vicwomg ,
Sorry for my ignorance, but where or how do I launch it with --prefer-ip? I tried running python3 app.py --prefer-ip from the windows command line after navigating to where I saved the pikaraoke1.2 folder and getting this message:

`C:\Program Files\pikaraoke-1.2>python3 app.py --prefer-ip
C:\Program Files\pikaraoke-1.2\app.py:617: SyntaxWarning: invalid escape sequence '\S'
  return os.path.join(os.path.dirname(__file__), ".venv\Scripts\yt-dlp.exe")
C:\Program Files\pikaraoke-1.2\app.py:642: SyntaxWarning: invalid escape sequence '\p'
  legacy_directory = os.path.expanduser("~\pikaraoke\songs")
C:\Program Files\pikaraoke-1.2\app.py:646: SyntaxWarning: invalid escape sequence '\p'
  return "~\pikaraoke-songs"
Traceback (most recent call last):
  File "C:\Program Files\pikaraoke-1.2\app.py", line 13, in <module>
    import cherrypy
ModuleNotFoundError: No module named 'cherrypy'`

I am not sure what I am doing wrong.

Thanks in advance for your help.

**Reply by vicwomg** [reply]
you should use the .bat launch script. Otherwise your python environment will not be activated: `./pikaraoke.bat --prefer-ip`

**Comment by lsibarra** [comment]
Hi @vicwomg,

Thank you so much for your help on this. It worked by running the command `./pikaraoke.bat --prefer-ip` from the command line. I tried running this setup in MacOS and keep getting issues with the flask-babel import locked_cached_property. Do you have any idea why I am getting that?

**Reply by vicwomg** [reply]
To be clear, on the mac the setup is ./setup.sh, and launch is ./pikaraoke.sh 

If either of those errored, the whole log would be good to see. Also OS version, etc.

**Comment by lsibarra** [comment]
Hi,
Yes I ran ./setup.sh and launched it with ./pikaraoke.sh from terminal. After doing that I get this:

```
ESIMAR@Mariels-MacBook-Pro pikaraoke % ./pikaraoke.sh
Traceback (most recent call last):
  File "app.py", line 14, in <module>
    import flask_babel
  File "/Users/Mariel/pikaraoke/.venv/lib/python3.8/site-packages/flask_babel/__init__.py", line 17, in <module>
    from flask.helpers import locked_cached_property
ImportError: cannot import name 'locked_cached_property' from 'flask.helpers' (/Users/Mariel/pikaraoke/.venv/lib/python3.8/site-packages/flask/helpers.py)
```

The macbook is running on MacOS Catalina version 10.15.7


Any idea?

**Reply by vicwomg** [reply]
I run it on a mac and cant reproduce that. Googling around there might be an incompatibility with Flask 3.0 which may have recently started making the rounds. 

Try changing requirements.txt line to from "Flask" to "Flask==2.3.3", then rerun `./setup.sh` and let me know if that fixes it. If so I might have to update the requirements file on master.

**Reply by vicwomg** [reply]
Actually, if that fails, try "Flask==2.2.5"

**Reply by lsibarra** [reply]
Ok.. let me try that now. Thanks.

**Reply by lsibarra** [reply]
Hi again @vicwomg,
I tried changing the requirements to this

`CherryPy
Flask==2.3.3
qrcode
psutil 
unidecode
requests
flask-paginate==2021.10.29
jinja2==3.0.3 # Avoid escape issue: https://github.com/sphinx-doc/sphinx/issues/10289
Babel==2.9.1
Flask-Babel==2.0.0
selenium
ffmpeg-python
yt-dlp`

and after rerunning ./setup.sh i got this error

`ESIMAR@Mariels-MacBook-Pro pikaraoke % ./setup.sh
Are you sure you want to setup PiKaraoke? (y/n): y
cat: /etc/os-release: No such file or directory
Client is not Debian-based. Skipping binary installation. Please install ffmpeg and chrome manually.

*** CREATING PYTHON VIRTUAL ENVIRONMENT ***

*** INSTALLING PYTHON DEPENDENCIES ***
Requirement already satisfied: CherryPy in ./.venv/lib/python3.12/site-packages (from -r requirements.txt (line 1)) (18.9.0)
Collecting Flask==2.3.3 (from -r requirements.txt (line 2))
  Obtaining dependency information for Flask==2.3.3 from https://files.pythonhosted.org/packages/fd/56/26f0be8adc2b4257df20c1c4260ddd0aa396cf8e75d90ab2f7ff99bc34f9/flask-2.3.3-py3-none-any.whl.metadata
  Using cached flask-2.3.3-py3-none-any.whl.metadata (3.6 kB)
Requirement already satisfied: qrcode in ./.venv/lib/python3.12/site-packages (from -r requirements.txt (line 3)) (7.4.2)
Requirement already satisfied: psutil in ./.venv/lib/python3.12/site-packages (from -r requirements.txt (line 4)) (5.9.7)
Requirement already satisfied: unidecode in ./.venv/lib/python3.12/site-packages (from -r requirements.txt (line 5)) (1.3.7)
Requirement already satisfied: requests in ./.venv/lib/python3.12/site-packages (from -r requirements.txt (line 6)) (2.31.0)
Requirement already satisfied: flask-paginate==2021.10.29 in ./.venv/lib/python3.12/site-packages (from -r requirements.txt (line 7)) (2021.10.29)
Requirement already satisfied: jinja2==3.0.3 in ./.venv/lib/python3.12/site-packages (from -r requirements.txt (line 8)) (3.0.3)
Requirement already satisfied: Babel==2.9.1 in ./.venv/lib/python3.12/site-packages (from -r requirements.txt (line 9)) (2.9.1)
Requirement already satisfied: Flask-Babel==2.0.0 in ./.venv/lib/python3.12/site-packages (from -r requirements.txt (line 10)) (2.0.0)
Requirement already satisfied: selenium in ./.venv/lib/python3.12/site-packages (from -r requirements.txt (line 11)) (4.16.0)
Requirement already satisfied: ffmpeg-python in ./.venv/lib/python3.12/site-packages (from -r requirements.txt (line 12)) (0.2.0)
Requirement already satisfied: yt-dlp in ./.venv/lib/python3.12/site-packages (from -r requirements.txt (line 13)) (2023.12.30)
Requirement already satisfied: Werkzeug>=2.3.7 in ./.venv/lib/python3.12/site-packages (from Flask==2.3.3->-r requirements.txt (line 2)) (3.0.1)
INFO: pip is looking at multiple versions of flask to determine which version is compatible with other requirements. This could take a while.
ERROR: Cannot install -r requirements.txt (line 2) and jinja2==3.0.3 because these package versions have conflicting dependencies.

The conflict is caused by:
    The user requested jinja2==3.0.3
    flask 2.3.3 depends on Jinja2>=3.1.2

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts

[notice] A new release of pip is available: 23.2.1 -> 23.3.2
[notice] To update, run: pip install --upgrade pip
ERROR: Python requirements.txt installation failed with error code: 0`

**Reply by vicwomg** [reply]
See my second note, try "Flask==2.2.5"

**Reply by lsibarra** [reply]
ok the Flask==2.2.5 worked at the ./setup.sh but after running ./pikaraoke.sh I am still getting this error

ESIMAR@Mariels-MacBook-Pro pikaraoke % ./pikaraoke.sh
Traceback (most recent call last):
  File "app.py", line 14, in <module>
    import flask_babel
  File "/Users/Mariel/pikaraoke/.venv/lib/python3.8/site-packages/flask_babel/__init__.py", line 17, in <module>
    from flask.helpers import locked_cached_property
ImportError: cannot import name 'locked_cached_property' from 'flask.helpers' (/Users/Mariel/pikaraoke/.venv/lib/python3.8/site-packages/flask/helpers.py)

**Reply by vicwomg** [reply]
Running out of ideas. Last two things:
- Delete the .venv directory and run setup again.
- Try installing python 3.11.X and delete .venv and try again

**Reply by lsibarra** [reply]
Hi @vicwomg 
Deleting the .venv folder and re-running the setup again worked. I was able to run ./pikaraoke.sh --prefer-ip without any problem. Thank you again for your help.

This means we still need to keep the Flask==2.2.5 version on the requirement.txt file for MacOS Catalina version 10.15.7.

Hope this help you get this working flawlessly.

I might reach out again because I am having issues with the raspberry pi 3 installation. The selenium python for the chromium webdriver not working. I will send you more detail tomorrow.

I am just testing the pikaraoke in different platform. Your work on this pikaraoke is really awesome as it can be installed in different platform.

Thanks for developing this.

Larry
